Zend/tests/type_declarations/variance/parent_in_class.phpt

```diff
@@ -0,0 +1,45 @@
+--TEST--
+Use of parent inside a class that has / has no parent
+--FILE--
+<?php
+
+// Illegal: A::parent is ill-defined
+class A {
+    public function method(parent $x) {}
+}
```

Comment:
Why is this allowed at all?

Reply:
I had originally added a compile-time check for this in 7.4, but reverted it again in https://github.com/php/php-src/commit/deb44d405eb27a6654ad9a57c1e5f641218b22a4 because it broke Mockery, which is used by Laravel. Maybe we can do this for PHP 8.

Reply:
might be a good idea to add this into deprecations for PHP-7.4